### PR TITLE
Fix 0.7 depwarns

### PR DESCRIPTION
**Pull request by ararslan**

### CODE REVIEW

File: src/ordered_set.jl [hunk 1]
```diff
@@ -30,9 +30,10 @@ pop!(s::OrderedSet, x, deflt) = pop!(s.dict, x, deflt) == deflt ? deflt : x
 delete!(s::OrderedSet, x) = (delete!(s.dict, x); s)
 
 getindex(x::OrderedSet,i::Int) = x.dict.keys[i]
-endof(x::OrderedSet) = endof(x.dict.keys)
+lastindex(x::OrderedSet) = lastindex(x.dict.keys)
 # Needed on 0.7 to mimic array indexing.
```
Comment (oxinabox) [inline src/ordered_set.jl]: Which line is this about?
`lastindex`  or `nextind`
we should either specify in the comment, or move the comment onto the same line as the line it is about

**Comment by ararslan** [comment]
> but it is failing on both CI for 0.7

Yep, still a work in progress for now.

**Comment by ararslan** [comment]
Still to fix:
```
/home/travis/.julia/v0.7/DataStructures/test/test_accumulator.jl ...
Accumulators: Error During Test at /home/travis/.julia/v0.7/DataStructures/test/test_accumulator.jl:116
  Test threw exception MethodError(convert, (Union{}, 1), 0x0000000000006a3d)
  Expression: counter((length(x) for x = s)) == counter(map(length, s))
  MethodError: convert(::Type{Union{}}, ::Int64) is ambiguous. Candidates:
    convert(::Type{T}, x::Number) where T<:Number in Base at number.jl:7
    convert(::Type{T}, x::Number) where T<:AbstractChar in Base at char.jl:166
    convert(::Type{T}, arg) where T<:VecElement in Base at sysimg.jl:76
  Possible fix, define
    convert(::Type{Union{}}, ::Number)
  Stacktrace:
   [1] setindex!(::Dict{Union{},Int64}, ::Int64, ::Int64) at ./dict.jl:374
   [2] setindex! at /home/travis/.julia/v0.7/DataStructures/src/accumulator.jl:50 [inlined]
   [3] inc! at /home/travis/.julia/v0.7/DataStructures/src/accumulator.jl:75 [inlined]
   [4] inc! at /home/travis/.julia/v0.7/DataStructures/src/accumulator.jl:76 [inlined]
   [5] counter(::Base.Generator{Array{String,1},typeof(length)}) at /home/travis/.julia/v0.7/DataStructures/src/accumulator.jl:22
   [6] macro expansion at /home/travis/.julia/v0.7/DataStructures/test/test_accumulator.jl:116 [inlined]
   [7] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/site/v0.7/Test/src/Test.jl:1022 [inlined]
   [8] top-level scope at /home/travis/.julia/v0.7/DataStructures/test/test_accumulator.jl:5
```
and
```
OrderedSet: Test Failed at /home/travis/.julia/v0.7/DataStructures/test/test_ordered_set.jl:140
  Expression: findfirst(isequal(7), s) == 4
   Evaluated: 7 == 4
Stacktrace:
 [1] macro expansion at /home/travis/.julia/v0.7/DataStructures/test/test_ordered_set.jl:140 [inlined]
 [2] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/site/v0.7/Test/src/Test.jl:1022 [inlined]
 [3] top-level scope at /home/travis/.julia/v0.7/DataStructures/test/test_ordered_set.jl:5
OrderedSet: Test Failed at /home/travis/.julia/v0.7/DataStructures/test/test_ordered_set.jl:141
  Expression: findfirst(isequal(2), s) == 0
   Evaluated: nothing == 0
Stacktrace:
 [1] macro expansion at /home/travis/.julia/v0.7/DataStructures/test/test_ordered_set.jl:141 [inlined]
 [2] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/site/v0.7/Test/src/Test.jl:1022 [inlined]
 [3] top-level scope at /home/travis/.julia/v0.7/DataStructures/test/test_ordered_set.jl:5
OrderedSet: Test Failed at /home/travis/.julia/v0.7/DataStructures/test/test_ordered_set.jl:199
  Expression: isequal(symdiff(OrderedSet([1, 2, 3, 4]), OrderedSet([2, 4, 5, 6])), OrderedSet([1, 3, 5, 6]))
   Evaluated: isequal([1, 3, 5, 6], OrderedSet{Int64}([1, 3, 5, 6]))
Stacktrace:
 [1] macro expansion at /home/travis/.julia/v0.7/DataStructures/test/test_ordered_set.jl:199 [inlined]
 [2] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/site/v0.7/Test/src/Test.jl:1022 [inlined]
 [3] top-level scope at /home/travis/.julia/v0.7/DataStructures/test/test_ordered_set.jl:5
```

**Comment by dfdx** [comment]
Huh, I've just fixed these issues locally, didn't even think somebody's working on it right now. Let me rebase and make a PR to your branch.

**Comment by ararslan** [comment]
This should be good to go now. Thanks so much for your help, @dfdx!

**Comment by ararslan** [comment]
Wow I completely forgot about this. Looks like all's well here now so I'll go ahead and merge and tag.

**Comment by kmsquire** [comment]
@ararslan, you probably saw this, but #377

**Comment by ararslan** [comment]
Whoops, my bad. I'll submit a PR to fix this momentarily...

**Pull request merged**